ostree-upload-server.py

```diff
@@ -40,18 +41,27 @@
 
 
 class UploadWebApp(Flask):
-    def __init__(self, import_name, users, repo, upload_counter,
-                 remote_push_adapter_map, task_queue):
+    # Allowed groups: lowercase, uppercase, numbers, and '-'
+    REPO_NAME_PATTERN = re.compile("^([A-Za-z\-0-9]+)+$")
+
+    def __init__(self, import_name, users, repo_prefix, upload_counter,
+                 remote_push_adapter_map, task_queue,
+                 enforce_target_repo=False):
         super(UploadWebApp, self).__init__(import_name)
         self._authenticator = Authenticator(users)
-        self._repo = repo
+        self._repo_prefix = repo_prefix
         self._upload_counter = upload_counter
         self._remote_push_adapter_map = remote_push_adapter_map
         self._task_queue = task_queue
+        self._enforce_target_repo = enforce_target_repo
+
+        self._active_repos = set([])
 
         # Sanity check
-        if not (os.path.exists(self._repo) and os.path.isdir(self._repo)):
-            raise RuntimeError("ERROR! Repo path '{}' is not valid!".format(self._repo))
+        if not (os.path.exists(self._repo_prefix) and os.path.isdir(self._repo_prefix)):
```

Comment:
There's no reason to call `exists` and `isdir` since the latter by necessity has to check for existence.